authorization/src/test.rs

```diff
@@ -33,62 +33,67 @@ fn make_auth(e: &Env, kp: &Keypair, args: Vec<RawVal>, function: &str) -> Signat
 fn test() {
     let env = Env::default();
     let contract_id = BytesN::from_array(&env, &[0; 32]);
-    env.register_contract(&contract_id, AuthContract);
+    env.register_contract(&contract_id, ExampleContract);
+    let client = ExampleContractClient::new(&env, contract_id);
 
-    // 1. set the admin
-    let admin = generate_keypair();
-    let admin_id = to_ed25519(&env, &admin);
-    let client = AuthContractClient::new(&env, contract_id);
+    // 1. Initialize contract by setting the admin.
+    let admin_kp = generate_keypair();
+    let admin_id = make_identifier(&env, &admin_kp);
     client.set_admin(&admin_id);
 
-    // 2. store data with user1's auth
-    let user1 = generate_keypair();
-    let user1_id = to_ed25519(&env, &user1);
-    let data = BigInt::from_u32(&env, 2);
+    // 2. Store a num for user1.
+    let user1_kp = generate_keypair();
+    let user1_id = make_identifier(&env, &user1_kp);
+    let num = BigInt::from_u32(&env, 2);
 
     let user1_nonce = client.nonce(&user1_id);
 
-    let mut args: Vec<RawVal> = Vec::new(&env);
-    args.push_back(user1_id.clone().into_val(&env));
-    args.push_back(user1_nonce.clone().into_val(&env));
-    args.push_back(data.clone().into_val(&env));
-    let auth = make_auth(&env, &user1, args, "save_data");
-    client.save_data(&auth, &user1_nonce, &data);
+    let auth = make_auth(
+        &env,
+        &user1_kp,
+        "save_num",
+        (&user1_id, &user1_nonce, &num).into_val(&env),
```

Comment:
The shorter tuple-to-vec notation is preferred over building the `Vec<RawVal>` by hand. The main advantage is it keeps the word `RawVal` out of contract code, and we should endeavor to reduce it wherever it shows up.